autoarray/plot/mat_wrap/wrap/wrap_base.py

```diff
@@ -0,0 +1,946 @@
+
+        This object wraps the following Matplotlib methods:
+
+        - plt.show: https://matplotlib.org/3.1.1/api/_as_gen/matplotlib.pyplot.show.html
+        - plt.savefig: https://matplotlib.org/3.1.1/api/_as_gen/matplotlib.pyplot.savefig.html
+
+        The default behaviour is the display the figure on the computer screen, as opposed to outputting to hard-disk
+        as a file.
+
+        Parameters
+        ----------
+        path : str
+            If the figure is output to hard-disk the path of the folder it is saved to.
+        filename : str
+            If the figure is output to hard-disk the filename used to save it.
+        format : str
+            The format of the output, 'show' displays on the computer screen, 'png' outputs to .png, 'fits' outputs to
+            `.fits` format.
+        bypass : bool
+            Whether to bypass the `plt.show` or `plt.savefig` methods, used when plotting a subplot.
+        """
+        self.path = path
+
+        if path is not None and path:
+            try:
+                os.makedirs(path)
+            except FileExistsError:
+                pass
+
+        self._filename = filename
+        self._auto_filename = None
+        self._format = format
+        self.bypass = bypass
+
+    @property
+    def filename(self):
+        if self._filename is not None:
+            return self._filename
+        return self._auto_filename
+
+    @property
+    def format(self) -> str:
+        if self._format is None:
+            return "show"
+        else:
+            return self._format
+
+    def to_figure(self, structure: abstract_structure.AbstractStructure):
+        """Output the figure, by either displaying it on the user's screen or to the hard-disk as a .png or .fits file.
+
+        Parameters
+        -----------
+        structure : abstract_structure.AbstractStructure
+            The 2D array of image to be output, required for outputting the image as a fits file.
+        """
+
+        if not self.bypass:
+            if self.format == "show":
+                plt.show()
+            elif self.format == "png":
+                plt.savefig(
+                    path.join(self.path, f"{self.filename}.png"), bbox_inches="tight"
+                )
+            elif self.format == "fits":
+                if structure is not None:
+                    structure.output_to_fits(
+                        file_path=path.join(self.path, f"{self.filename}.fits"),
+                        overwrite=True,
+                    )
+
+    def subplot_to_figure(self):
+        """Output a subhplot figure, either as an image on the screen or to the hard-disk as a .png or .fits file."""
+        if self.format == "show":
+            plt.show()
+        elif self.format == "png":
+            plt.savefig(
+                path.join(self.path, f"{self.filename}.png"), bbox_inches="tight"
+            )
+
+
+class AbstractMatWrapColored:
+
+    config_dict = None

```

Comment:
Why is this config dict defined here? Unless you're intending to cache global state it's probably not a good idea

Reply:
Lint errors. Hopefully I can get rid of these config dicts...

Reply:
You might be able to get away with declaring the type

config_dict: dict

Else think about ensuring the dict is instantiated in the correct constructor

although none of this matters if you get rid of it